encoder/vaapiencoder_h264.cpp

```diff
@@ -939,7 +939,10 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
         m_reorderFrameList.push_front(picture);
         m_curFrameNum++;
         m_reorderState = VAAPI_ENC_REORD_DUMP_FRAMES;
-    }else if (m_frameIndex % (m_numBFrames + 1) != 0) {
+    } else if ((m_frameIndex % (m_numBFrames + 1) != 0)
+            && (m_frameIndex + 1 != intraPeriod())
+            && (m_frameIndex + 1 != keyFramePeriod())) {
+        // If next frame will be I or Idr, set this frame to P instead of B.
```

Comment:
Only when next frame is IDR, need to set this frame to be P frame, because only IDR( not I frame ) need to refresh DPB buffer.